app/src/main/java/com/jerboa/ui/components/blocks/BlockedCommunityViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.jerboa.ui.components.blocks
+
+import android.content.Context
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.CreationExtras
+import com.jerboa.api.API
+import com.jerboa.api.ApiState
+import com.jerboa.api.toApiState
+import com.jerboa.feat.showBlockCommunityToast
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockCommunity
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockCommunityResponse
+import it.vercruysse.lemmyapi.v0x19.datatypes.CommunityId
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class BlockedCommunityViewModel(
+    private val communityId: CommunityId,
```

Comment:
You only need these input params for the viewmodels, if an `init` function is required (for example, to initially fetch a person's profile). Those aren't necessary here. Just pass in the IDs to the block functions, and remove the factories.